docker/reference/reference.go

```diff
@@ -102,13 +107,13 @@ func WithTag(name Named, tag string) (NamedTagged, error) {
 
 // WithDigest combines the name from "name" and the digest from "digest" to form
 // a reference incorporating both the name and the digest.
-func WithDigest(name Named, digest digest.Digest) (Canonical, error) {
-	r, err := distreference.WithDigest(name, digest)
-	if err != nil {
-		return nil, err
-	}
-	return &canonicalRef{namedRef{r}}, nil
-}
+//func WithDigest(name Named, digest digest.Digest) (Canonical, error) {
```

Comment:
> I am very confused how this happens. docker/distribution/reference imports as "github.com/opencontainers/go-digest", so why does Golang decide to rewrite using the /vendor path (and then prohibit naming that path, per golang/go#12111 )?

@mtrmac this should be fully explained here https://docs.google.com/document/d/1Bz5-UB7g2uPBdOx-rw5t9MxJwkfpx90cqG9AFL0JAYo/edit (unless there's a bug somewhere...)